.github/workflows/ci.yml

```diff
@@ -455,3 +455,93 @@ jobs:
           # is merged and the package becomes generally available.
           SELINUX: Permissive
         run: vagrant up --provision-with=selinux,install-runc,test-cri
+
+  integration-macos:
+    name: macOS Integration
+    runs-on: macos-10.15
+    timeout-minutes: 30
+    needs: [project, linters, protos, man]
+
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.13.15'
+
+      - name: Set env
+        shell: bash
+        run: |
+          echo "::set-env name=GOPATH::${{ github.workspace }}"
+          echo "::add-path::${{ github.workspace }}/src/github.com/containerd/containerd/bin"
+
+      - name: Checkout containerd
+        uses: actions/checkout@v2
+        with:
+          path: src/github.com/containerd/containerd
+      - name: Checkout runu
+        uses: actions/checkout@v2
+        with:
+          repository: ukontainer/runu
+          path: src/github.com/ukontainer/runu
+
+      - name: Install Build Deps
+        shell: bash
+        run: |
+          cd src/github.com/containerd/containerd
+          script/setup/install-dev-tools
+      - name: Install runtime
+        shell: bash
+        run: |
+          SHIM_COMMIT=v0.8.2
+          GO111MODULE=on go get -v github.com/ukontainer/runu@"${SHIM_COMMIT}"
+          bindir="$(pwd)/bin/"
+          cd src/github.com/ukontainer/runu
+          git fetch --tags origin "${SHIM_COMMIT}"
+          git checkout "${SHIM_COMMIT}"
+          GO111MODULE=on go build -o "${bindir}/containerd-shim-darwin-v1" ./cmd/containerd-shim-darwin-v1
+      - name: Binaries
+        shell: bash
+        run: |
+          set -o xtrace
+          cd src/github.com/containerd/containerd
```

Comment:
nit: `working-directory: src/github.com/containerd/containerd`
(same below and above)

Reply:
added multiple `working-directory`.
defaluts helps indeed, but some steps before checkouting containerd will fail so, I didn't.

thanks.

Reply:
I was wrong; If the checkouting containerd is the earliest, I can use the `defaults` clause to simplify.  I've updated the ci.yml (and rebased).